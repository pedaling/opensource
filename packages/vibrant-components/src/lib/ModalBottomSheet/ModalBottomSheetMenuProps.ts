import type { ReactElement } from 'react';
import { propVariant, withVariation } from '@vibrant-ui/core';

export type ModalBottomSheetMenuProps = {
  icon?: ReactElement;
  title: string;
  onClick?: () => void;
  rightContents?: () => ReactElement;
  disabled?: boolean;
};

export const withModalBottomSheetMenuVariation = withVariation<ModalBottomSheetMenuProps>('ModalBottomSheetMenu')(
  propVariant({
    props: [
      {
        name: 'disabled',
        default: false,
      },
    ],
    variants: {
      false: {
        color: 'onView1',
      } as const,
      true: {
        color: 'onView2',
      } as const,
    },
  })
);
