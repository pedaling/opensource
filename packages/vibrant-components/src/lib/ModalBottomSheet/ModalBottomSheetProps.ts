import type { ReactElementChild } from '@vibrant-ui/core';
import { propVariant, withVariation } from '@vibrant-ui/core';
import type { Either } from '@vibrant-ui/utils';

export type ModalBottomSheetProps = Either<
  { open: boolean },
  {
    defaultOpen: boolean;
    renderOpener: (_: { open: () => void; isOpen: boolean }) => ReactElementChild;
  }
> & {
  title?: string;
  subtitle?: string;
  size?: 'lg' | 'md';
  testId?: string;
  renderContents?: (_: { close: () => void }) => ReactElementChild;
  onClose?: () => void;
  showCloseButton?: boolean;
} & (
    | {
        primaryButtonOptions: ButtonOptions;
        secondaryButtonOptions?: ButtonOptions;
        subButtonOptions?: never;
      }
    | {
        primaryButtonOptions: ButtonOptions;
        secondaryButtonOptions?: never;
        subButtonOptions?: Omit<ButtonOptions, 'loading'>;
      }
    | {
        primaryButtonOptions?: never;
        secondaryButtonOptions?: never;
        subButtonOptions?: never;
      }
  );

export type ButtonOptions = {
  text: string;
  disabled?: boolean;
  loading?: boolean;
  onClick?: (_: { close: () => void }) => void;
};

export const withModalBottomSheetVariation = withVariation<ModalBottomSheetProps>('ModalBottomSheet')(
  propVariant({
    props: [
      {
        name: 'size',
        default: 'md',
      },
    ],
    variants: {
      md: {
        desktopModalWidth: 480,
      },
      lg: {
        desktopModalWidth: 760,
      },
    },
  })
);
