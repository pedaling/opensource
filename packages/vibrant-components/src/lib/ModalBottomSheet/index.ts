export { ModalBottomSheet } from './ModalBottomSheet';
export { ModalBottomSheetProps } from './ModalBottomSheetProps';
