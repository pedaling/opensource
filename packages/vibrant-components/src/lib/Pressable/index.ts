export { Pressable } from './Pressable';
export type { PressableProps } from './PressableProp';
export { getOpacity } from './getOpacity';
