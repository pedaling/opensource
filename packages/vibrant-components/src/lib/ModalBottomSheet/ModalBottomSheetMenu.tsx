import { PressableBox } from '@vibrant-ui/core';
import { Body } from '../Body';
import { HStack } from '../HStack';
import { withModalBottomSheetMenuVariation } from './ModalBottomSheetMenuProps';

export const ModalBottomSheetMenu = withModalBottomSheetMenuVariation(
  ({ rightContents, IconComponent, onClick, title, color }) => (
    <PressableBox onClick={onClick} flexDirection="row" alignItems="center" py={12}>
      <HStack alignVertical="center" width="100%" spacing={8}>
        {IconComponent && <IconComponent size={20} fill={color} />}
        <Body level={1} color={color} flex={1}>
          {title}
        </Body>
        {rightContents?.()}
      </HStack>
    </PressableBox>
  )
);
