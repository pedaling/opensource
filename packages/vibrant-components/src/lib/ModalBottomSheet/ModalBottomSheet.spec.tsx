import { fireEvent, waitFor, waitForElementToBeRemoved } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Box, PortalRootProvider } from '@vibrant-ui/core';
import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { BrowserTesting, createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { Pressable } from '../Pressable';
import { ModalBottomSheet } from './ModalBottomSheet';

describe('<ModalBottomSheet />', () => {
  const { render } = createReactRenderer(children => <PortalRootProvider zIndex={1}>{children}</PortalRootProvider>);
  let renderer: ReactRenderer;

  describe('when defaultOpen is false', () => {
    beforeEach(() => {
      renderer = render(
        <ModalBottomSheet
          defaultOpen={false}
          renderOpener={({ open }) => <Pressable data-testid="opener" onClick={open} />}
        />
      );
    });

    it('should not open the content', () => {
      expect(renderer.queryByRole('dialog')).toBeFalsy();
    });

    describe('after clicking the opener button', () => {
      beforeEach(done => {
        waitFor(() => userEvent.click(renderer.getByTestId('opener'))).then(done);
      });

      it('should open the content', () => {
        expect(renderer.queryByRole('dialog')).toBeTruthy();
      });
    });
  });

  describe('when defaultOpen is true', () => {
    let mockOnClose: jest.Mock<any, any>;

    beforeEach(() => {
      mockOnClose = jest.fn();

      renderer = render(<ModalBottomSheet defaultOpen={true} onClose={mockOnClose} renderOpener={() => null} />);
    });

    it('should open the content', () => {
      expect(renderer.queryByRole('dialog')).toBeTruthy();
    });

    describe('after clicking the close button', () => {
      beforeEach(async () => {
        fireEvent.click(renderer.getByRole('button', { name: 'Close' }));

        await waitForElementToBeRemoved(() => renderer.queryByRole('dialog'));
      });

      it.skip('should call onClose function', () => {
        expect(mockOnClose).toBeCalled();
      });

      it.skip('should close the content', () => {
        expect(renderer.queryByRole('dialog')).toBeFalsy();
      });
    });
  });

  describe('when renderContents is provided', () => {
    beforeEach(() => {
      renderer = render(<ModalBottomSheet open={true} renderContents={() => <Box data-testid="contents" />} />);
    });

    it('should render returned element of renderContents', () => {
      expect(renderer.queryByTestId('contents')).toBeTruthy();
    });
  });

  describe('when title is provided', () => {
    beforeEach(() => {
      renderer = render(<ModalBottomSheet open={true} title="title" />);
    });

    it('should show title text', () => {
      expect(renderer.queryByRole('heading', { name: 'title' })).toBeTruthy();
    });
  });

  describe('when subtitle is provided', () => {
    beforeEach(() => {
      renderer = render(<ModalBottomSheet open={true} subtitle="subtitle" />);
    });

    it('should show subtitle text', () => {
      expect(renderer.queryByText('subtitle')).toBeTruthy();
    });
  });

  describe('when primaryButtonOptions are provided', () => {
    let mockOnPrimaryButtonClick: jest.Mock<any, any>;

    beforeEach(() => {
      mockOnPrimaryButtonClick = jest.fn();

      renderer = render(
        <ModalBottomSheet
          defaultOpen={true}
          primaryButtonOptions={{
            text: 'primaryButtonText',
            onClick: mockOnPrimaryButtonClick,
          }}
          secondaryButtonOptions={undefined}
          subButtonOptions={undefined}
          renderOpener={({ open }) => <Pressable onClick={open} />}
        />
      );
    });

    it('should show the primary button', () => {
      expect(renderer.queryByRole('button', { name: 'primaryButtonText' })).toBeTruthy();
    });

    describe('after clicking the primary button', () => {
      beforeEach(done => {
        waitFor(() => userEvent.click(renderer.getByRole('button', { name: 'primaryButtonText' }))).then(done);
      });

      it('should call OnPrimaryButtonClick function', () => {
        expect(mockOnPrimaryButtonClick).toBeCalled();
      });
    });
  });

  describe('when primaryButtonOptions, secondaryButtonOptions are provided', () => {
    let mockOnPrimaryButtonClick: jest.Mock<any, any>;
    let mockOnSecondaryButtonClick: jest.Mock<any, any>;

    beforeEach(() => {
      mockOnPrimaryButtonClick = jest.fn();

      mockOnSecondaryButtonClick = jest.fn();

      renderer = render(
        <ModalBottomSheet
          defaultOpen={true}
          primaryButtonOptions={{
            text: 'primaryButtonText',
            onClick: mockOnPrimaryButtonClick,
          }}
          secondaryButtonOptions={{
            text: 'secondaryButtonText',
            onClick: mockOnSecondaryButtonClick,
          }}
          subButtonOptions={undefined}
          renderOpener={({ open }) => <Pressable onClick={open} />}
        />
      );
    });

    it('should show the primary button', () => {
      expect(renderer.queryByRole('button', { name: 'primaryButtonText' })).toBeTruthy();
    });

    describe('after clicking the primary button', () => {
      beforeEach(done => {
        waitFor(() => userEvent.click(renderer.getByRole('button', { name: 'primaryButtonText' }))).then(done);
      });

      it('should call OnPrimaryButtonClick function', () => {
        expect(mockOnPrimaryButtonClick).toBeCalled();
      });
    });

    it('should show the secondary button', () => {
      expect(renderer.queryByRole('button', { name: 'secondaryButtonText' })).toBeTruthy();
    });

    describe('after clicking the secondary button', () => {
      beforeEach(done => {
        waitFor(() => userEvent.click(renderer.getByRole('button', { name: 'secondaryButtonText' }))).then(done);
      });

      it('should call OnSecondaryButtonClick function', () => {
        expect(mockOnSecondaryButtonClick).toBeCalled();
      });
    });
  });

  describe('when primaryButtonOptions, subButtonOptions are provided', () => {
    let mockOnPrimaryButtonClick: jest.Mock<any, any>;
    let mockOnSubButtonClick: jest.Mock<any, any>;

    beforeEach(() => {
      mockOnPrimaryButtonClick = jest.fn();

      mockOnSubButtonClick = jest.fn();

      renderer = render(
        <ModalBottomSheet
          defaultOpen={true}
          primaryButtonOptions={{
            text: 'primaryButtonText',
            onClick: mockOnPrimaryButtonClick,
          }}
          secondaryButtonOptions={undefined}
          subButtonOptions={{
            text: 'subButtonText',
            onClick: mockOnSubButtonClick,
          }}
          renderOpener={({ open }) => <Pressable onClick={open} />}
        />
      );
    });

    it('should show the primary button', () => {
      expect(renderer.queryByRole('button', { name: 'primaryButtonText' })).toBeTruthy();
    });

    describe('after clicking the primary button', () => {
      beforeEach(done => {
        waitFor(() => userEvent.click(renderer.getByRole('button', { name: 'primaryButtonText' }))).then(done);
      });

      it('should call OnPrimaryButtonClick function', () => {
        expect(mockOnPrimaryButtonClick).toBeCalled();
      });
    });

    it('should show the sub button', () => {
      expect(renderer.queryByRole('button', { name: 'subButtonText' })).toBeTruthy();
    });

    describe('after clicking the sub button', () => {
      beforeEach(done => {
        waitFor(() => userEvent.click(renderer.getByRole('button', { name: 'subButtonText' }))).then(done);
      });

      it('should call onSubButtonClick function', () => {
        expect(mockOnSubButtonClick).toBeCalled();
      });
    });
  });

  describe('given mobile viewport width', () => {
    beforeEach(async () => {
      await BrowserTesting.resize(639, 768);

      renderer = render(<ModalBottomSheet open={true} renderContents={() => null} />);
    });

    it('match snapshot', async () => {
      await waitFor(() => expect(renderer.getByRole('dialog').style.transform).toBe(''));

      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('given viewport width larger than mobile', () => {
    beforeEach(async () => {
      await BrowserTesting.resize(1024, 768);

      renderer = render(<ModalBottomSheet open={true} renderContents={() => null} />);
    });

    it('match snapshot', async () => {
      await waitFor(() => expect(renderer.getByRole('dialog').style.opacity).toBe('1'));

      expect(renderer.container).toMatchSnapshot();
    });
  });
});
