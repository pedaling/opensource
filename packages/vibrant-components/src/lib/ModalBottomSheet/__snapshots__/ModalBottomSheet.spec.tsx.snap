// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalBottomSheet /> given mobile viewport width match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  z-index: 1;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: fixed;
  z-index: 100;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: #000000b2;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-top: 120px;
  padding-bottom: 0;
}

@media screen and (min-width: 640px) {
  .emotion-2 {
    padding-top: 40px;
    padding-bottom: 40px;
  }
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: absolute;
  cursor: default;
  overflow: hidden;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  width: auto;
  height: auto;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  margin-left: auto;
  margin-right: auto;
  margin-top: auto;
  margin-bottom: 0;
  padding-top: 24px;
  padding-bottom: 20px;
  width: 100%;
  max-height: 648px;
  background-color: #ffffff;
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

@media screen and (min-width: 640px) {
  .emotion-5 {
    margin-bottom: auto;
    padding-top: 36px;
    padding-bottom: 36px;
    width: 480px;
    max-height: 688px;
    border-bottom-left-radius: 20px;
    border-bottom-right-radius: 20px;
  }
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  position: relative;
  padding-left: 20px;
  padding-right: 20px;
}

@media screen and (min-width: 640px) {
  .emotion-6 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  background-color: inherit;
  border-width: 0;
  padding: 0;
  overflow: hidden;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  z-index: 0;
  margin-left: auto;
  margin-top: 0;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  width: auto;
  height: auto;
  padding-left: 12px;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  width: 24px;
  height: 24px;
  fill: #0c0c0c;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  overflow-y: auto;
  margin-top: 16px;
}

@media screen and (min-width: 640px) {
  .emotion-11 {
    margin-top: 24px;
  }
}

<div>
  <div
    class="emotion-0"
    id="portal-root"
  >
    <div
      class="emotion-1"
      style="opacity: 0;"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
          tabindex="0"
        >
          <span
            class="emotion-4"
            style="opacity: 1;"
          />
        </div>
        <div
          class="emotion-5"
          role="dialog"
          style="transform: translate3d(0,768px,0);"
        >
          <div
            class="emotion-6"
          >
            <button
              aria-label="Close"
              class="emotion-7"
              type="button"
            >
              <span
                class="emotion-8"
                style="opacity: 1;"
              >
                <svg
                  class="emotion-9"
                  viewBox="0 0 24 24"
                >
                  <path
                    class="emotion-10"
                    d="m13.75 12 6.35-6.35c.1-.1.1-.25 0-.35l-1.4-1.4c-.1-.1-.25-.1-.35 0L12 10.25l-6.35-6.4c-.1-.1-.25-.1-.35 0l-1.4 1.4c-.1.1-.1.25 0 .35l6.35 6.4-6.4 6.35c-.1.1-.1.25 0 .35l1.4 1.4c.1.1.25.1.35 0l6.4-6.35 6.35 6.35c.1.1.25.1.35 0l1.4-1.4c.1-.1.1-.25 0-.35L13.75 12Z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ModalBottomSheet /> given viewport width larger than mobile match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  z-index: 1;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  overflow-y: auto;
  position: fixed;
  z-index: 100;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: #000000b2;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-top: 120px;
  padding-bottom: 0;
}

@media screen and (min-width: 640px) {
  .emotion-2 {
    padding-top: 40px;
    padding-bottom: 40px;
  }
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: absolute;
  cursor: default;
  overflow: hidden;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  width: auto;
  height: auto;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  margin-left: auto;
  margin-right: auto;
  margin-top: auto;
  margin-bottom: 0;
  padding-top: 24px;
  padding-bottom: 20px;
  width: 100%;
  max-height: 648px;
  background-color: #ffffff;
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

@media screen and (min-width: 640px) {
  .emotion-5 {
    margin-bottom: auto;
    padding-top: 36px;
    padding-bottom: 36px;
    width: 480px;
    max-height: 688px;
    border-bottom-left-radius: 20px;
    border-bottom-right-radius: 20px;
  }
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  position: relative;
  padding-left: 20px;
  padding-right: 20px;
}

@media screen and (min-width: 640px) {
  .emotion-6 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  background-color: inherit;
  border-width: 0;
  padding: 0;
  overflow: hidden;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  z-index: 0;
  margin-left: auto;
  margin-top: 0;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  width: auto;
  height: auto;
  padding-left: 12px;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  width: 24px;
  height: 24px;
  fill: #0c0c0c;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  overflow-y: auto;
  margin-top: 16px;
}

@media screen and (min-width: 640px) {
  .emotion-11 {
    margin-top: 24px;
  }
}

<div>
  <div
    class="emotion-0"
    id="portal-root"
  >
    <div
      class="emotion-1"
      style="opacity: 0;"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
          tabindex="0"
        >
          <span
            class="emotion-4"
            style="opacity: 1;"
          />
        </div>
        <div
          class="emotion-5"
          role="dialog"
          style="opacity: 1;"
        >
          <div
            class="emotion-6"
          >
            <button
              aria-label="Close"
              class="emotion-7"
            >
              <span
                class="emotion-8"
                style="opacity: 1;"
              >
                <svg
                  class="emotion-9"
                  viewBox="0 0 24 24"
                >
                  <path
                    class="emotion-10"
                    d="m13.75 12 6.35-6.35c.1-.1.1-.25 0-.35l-1.4-1.4c-.1-.1-.25-.1-.35 0L12 10.25l-6.35-6.4c-.1-.1-.25-.1-.35 0l-1.4 1.4c-.1.1-.1.25 0 .35l6.35 6.4-6.4 6.35c-.1.1-.1.25 0 .35l1.4 1.4c.1.1.25.1.35 0l6.4-6.35 6.35 6.35c.1.1.25.1.35 0l1.4-1.4c.1-.1.1-.25 0-.35L13.75 12Z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
